dagshub/data_engine/model/datasource.py

```diff
@@ -120,6 +154,17 @@ def all(self) -> "QueryResult":
         self._check_preprocess()
         return self._source.client.get_datapoints(self)
 
+    def select(self, *selected: Field):
+        """
+        Choose which columns will appear on the query result, what their names will be (alias) and from what time.
+        example:
+        t = int((datetime.datetime.now()-datetime.timedelta(hours=24)).timestamp())
+        q1 = (ds["episode"] > 5).select(Field("episode", as_of_time=t, alias="episode_asof_t"), Field("size"))
```

Comment:
Python best practice: never use naive datetimes.
```suggestion
        Choose which columns will appear on the query result, what their names will be (alias) and from what time.
        example:
        from datetime import datetime, timedelta, timezone
        yesterday = datetime.now(timezone.utc) - timedelta(days=1)
        q1 = (ds["episode"] > 5).select(Field("episode", as_of_time=yesterday, alias="episode_yesterday"), Field("size"))
```

I'd go even as far as to say that we should explicitly not allow naive datetimes, because they can lead to unexpected results when ran on machines in different timezones.
You can read up a bit on it [here](https://realpython.com/python-datetime/#comparing-naive-and-aware-python-datetime-instances)

Reply:
Also question regarding the whole select thing, backend included: If I apply a select, does it ONLY show the selected fields, or are they added on top of all the other basic fields?

Whatever the answer, it needs to be written out here